src/TreeView/TreeView.tsx

```diff
@@ -463,23 +474,15 @@ const Item = React.forwardRef<HTMLElement, TreeViewItemProps>(
               contentVisibility: containIntrinsicSize ? 'auto' : undefined,
               containIntrinsicSize,
             }}
-            onClick={event => {
-              if (onSelect) {
-                onSelect(event)
-              } else {
-                toggle(event)
-              }
-            }}
-            onAuxClick={event => {
-              if (onSelect && event.button === 1) {
-                onSelect(event)
-              }
-            }}
```

Comment:
Hey @siddharthkp - How do you feel about this change? I saw that all storybooks worked when i moved these events to the parent.

Reply:
Not confident 😓 https://github.com/primer/react/pull/3853#issuecomment-1794546357

I'm wondering if we're keeping once instance of eslint-ignore of these rules, should we keep the other one as well.

Bonus: This component is very detailed, should we write some tests instead of manual tests to bring confidence?